src/range_proof.rs

```diff
@@ -1,18 +1,24 @@
 use bulletproofs::r1cs::{ConstraintSystem, LinearCombination, R1CSError};
 use curve25519_dalek::scalar::Scalar;
 
+use crate::value::SignedInteger;
+
 /// Enforces that the quantity of v is in the range [0, 2^n).
 pub fn range_proof<CS: ConstraintSystem>(
     cs: &mut CS,
     mut v: LinearCombination,
-    v_assignment: Option<u64>,
+    v_assignment: Option<SignedInteger>,
     n: usize,
 ) -> Result<(), R1CSError> {
     let mut exp_2 = Scalar::one();
     for i in 0..n {
         // Create low-level variables and add them to constraints
         let (a, b, o) = cs.allocate(|| {
-            let q: u64 = v_assignment.ok_or(R1CSError::MissingAssignment)?;
+            // let q: u64 = v_assignment.ok_or(R1CSError::MissingAssignment)?;
+            let q: u64 = v_assignment
+                .ok_or(R1CSError::MissingAssignment)?
+                .to_u64()
+                .ok_or(R1CSError::MissingAssignment)?;
```

Comment:
i think the more apt error is `::InconsistentAssignment`.

Reply:
there's only `R1CSError::MissingAssignment`, but we could define a spacesuit `InconsistentAssignment` error?

Reply:
Oh.. we are using errors from r1cs API. What about using MissingAssignment for now and add a comment “TBD: change to InvalidAssignment”? R1CS api is not set in stone, but we already one subproject deep in our troubles.

Reply:
:) that sounds good to me!